public/tabs/history/chat_history_page.tsx

```diff
@@ -96,14 +100,29 @@ export const ChatHistoryPage: React.FC<ChatHistoryPageProps> = React.memo((props
     [searchName]
   );
 
-  useEffect(() => {
-    if (props.shouldRefresh) services.conversations.reload();
+  useUpdateEffect(() => {
+    if (!props.shouldRefresh) {
+      return;
+    }
+    services.conversations.reload();
+    return () => {
+      services.conversations.abortController?.abort();
+    };
   }, [props.shouldRefresh, services.conversations]);
 
   useEffect(() => {
     services.conversations.load(bulkGetOptions);
+    const subscription = services.dataSource
+      .getDataSourceId$()
```

Comment:
Would it be possible that `dataSourceId$` emit the same value one after another? In such case, I think we don't want to reload.

Reply:
In current implementation, the same value will be emit after default data source change. Do you have any suggestions about that? How about add a `distinctUntilChanged`?